src/sync/installation.ts

```diff
@@ -209,14 +209,221 @@ export const isNotFoundError = (
+			job,
+			edges,
+			task,
+			repositoryId,
+			logger,
+		);
+
+		statsd.increment(metricTaskStatus.complete, [`type: ${nextTask.task}`]);
+
+	} catch (err) {
+		const rateLimit = Number(err?.headers?.["x-ratelimit-reset"]);
+		const delay = Math.max(Date.now() - rateLimit * 1000, 0);
+
+		if (delay) {
+			// if not NaN or 0
+			logger.info({ delay, job, task: nextTask }, `Delaying job for ${delay}ms`);
+			queues.installation.add(job.data, { delay });
+			return;
+		}
+
+		if (String(err).includes("connect ETIMEDOUT")) {
+			// There was a network connection issue.
+			// Add the job back to the queue with a 5 second delay
+			logger.warn({ job, task: nextTask }, "ETIMEDOUT error, retrying in 5 seconds");
+			queues.installation.add(job.data, { delay: 5000 });
+			return;
+		}
+
+		if (
+			String(err.message).includes(
+				"You have triggered an abuse detection mechanism"
+			)
+		) {
+			// Too much server processing time, wait 60 seconds and try again
+			logger.warn({ job, task: nextTask }, "Abuse detection triggered. Retrying in 60 seconds");
+			queues.installation.add(job.data, { delay: 60000 });
+			return;
+		}
+
+		// Continue sync when a 404/NOT_FOUND is returned
+		if (isNotFoundError(err, job, nextTask, logger)) {
+			const edgesLeft = []; // No edges left to process since the repository doesn't exist
+			await updateJobStatus(queues, job, edgesLeft, task, repositoryId, logger);
+			return;
+		}
+
+		if (await booleanFlag(BooleanFlags.CONTINUE_SYNC_ON_ERROR, false, jiraHost)) {
+
+			// TODO: add the jiraHost to the logger with logger.child()
+			const host = subscription.jiraHost || "none";
+			logger.warn({ job, task: nextTask, err, jiraHost: host }, "Task failed, continuing with next task");
+
+			// marking the current task as failed
+			await subscription.updateRepoSyncStateItem(nextTask.repositoryId, getStatusKey(nextTask.task as TaskType), "failed");
+
+			statsd.increment(metricTaskStatus.failed, [`type: ${nextTask.task}`]);
+
+			// queueing the job again to pick up the next task
+			queues.installation.add(job.data);
+		} else {
+
+			await subscription.update({ syncStatus: "FAILED" });
+
+			// TODO: add the jiraHost to the logger with logger.child()
+			const host = subscription.jiraHost || "none";
+			logger.warn({ job, task: nextTask, err, jiraHost: host }, "Sync failed");
+
+			job.sentry.setExtra("Installation FAILED", JSON.stringify(err, null, 2));
+			job.sentry.captureException(err);
+
+			statsd.increment(metricSyncStatus.failed);
+
+			throw err;
+		}
+	}
+}
+
 // TODO: type queues
 export const processInstallation =
 	(app: Application, queues) =>
-		async (job, logger: LoggerWithTarget): Promise<void> => {
-			const { installationId, jiraHost } = job.data;
+		async (job, rootLogger: LoggerWithTarget): Promise<void> => {
+			const {installationId, jiraHost} = job.data;
+
+			let logger: LoggerWithTarget;
+			if (await booleanFlag(BooleanFlags.SPLIT_PROCESS_INSTALLATION_FUNCTION, false, jiraHost)) {
```

Comment:
The default for this flag should be `true` otherwise we'll have to manually change this if we want to hit the new code locally.

- [ ] change false to true

Reply:
According to yesterday's discussion it should be false: https://atlassian.slack.com/archives/C02KGV1MELE/p1635740495053400

Reply:
Interesting. Michel made me set all mine to true 🤔

Reply:
We can use LaunchDarkly development environment locally, we need to copy LD key to `.env` file for that. That's how I usually test